### DISCUSSION TITLE
VTT formatted output for wav2letter / flashlight-asr  (compatible with Able Player)

### DISCUSSION THREAD

**Discussion by dev2devportal** [flashlight/flashlight#737] [Q&A]
Greetings,
We have wav2letter / flashlight-asr working fine ingesting .wav audio files, and creating human readable high quality text transcripts.
Now we're trying to figure out how to either simultaneously create .vtt file, convert the created file to .vtt, or it take the same audio and create a .vtt file for closed captions to be imported and working in Able Player.
We can't find any information on this format with w2l / f-asr. We do see about 10 other tools out there, but we were hoping to have the higher quality accuracy of f-asr, rather than run a third tool (we're also running Vosk + Jigasi in Jitsi for real-time closed captions).
Does anyone have any information, configuration file that can make this possible without our having to create a whole definition file from scratch?
Thank you kindly for any guidance you may be able to provide.

**Comment by tlikhomanenko** [comment]
cc @jacobkahn @vineelpratap

**Reply by dev2devportal** [reply]
Greetings, a gentle bump to see if any suggestions please. Thanks kindly.

**Reply by dev2devportal** [reply]
Right now the developers in desperation are kludging together multiple scripts, but I am still holding out hope there is a more elegant approach than this.

AV recording created by Jitsi + Jibri, creating AV file.
Real-time closed captions created by Jitsi + Jigasi + Vosk server (no text file saved).
After meeting is over, the jibri created AV file is copied to wav2letter (w2l) / flashlight-asr (fasr) server storage.
ffmpeg separates audio from AV file and creates .wav file.
w2l/fasr ingest .wav file and creates high quality human readable text transcript as output .txt (higher quality tan Vosk can create).
The above is working as-is.

Now the kludge to also create .vtt and .srt formatted text files, so that a .vtt file can be imported with AV file for playback in Able Player with closed captions in sync:

On the w2l/fasr server, the AV file is converted to .srt through using nodejs with express to create server where we will upload our text file. Then we will convert the text file to SRT using txt2sub.js.
Then we run nwoltman srt-to-vtt script to convert the .srt to .vtt format.
Then the .txt, .srt, and .vtt files are moved to long-term storage and links stored in metadb.
AV file copied to long-term AV file storage and links stored in metadb.
Then .vtt is imported into Able Player with the associated AV, creating closed captions in the Able Player.
Then web link is created for end-user to view online video that now includes the synchronized closed captions.
This is what we are now working on, and have each of the pieces working, and we're about to try to merge into a complete (kludged) process.

I am really hoping folks here have suggestions formuch more elegant and efficient ways to do this (without having to custom code everything from scratch).

Appreciate any suggestions.

**Reply by jacobkahn** [reply]
@dev2devportal — we use [libsndfile](http://www.mega-nerd.com/libsndfile/) to load an decode audio -- it unfortunately [doesn't support](https://libsndfile.github.io/libsndfile/formats.html) the formats in question.

To get this to work, you'd need to [add code to decode audio in that format here](https://github.com/flashlight/flashlight/blob/v0.3/flashlight/app/asr/data/Sound.cpp). Let me know if I can answer more questions.
